p2p/transport/webrtc/listener.go

```diff
@@ -0,0 +1,292 @@
+package libp2pwebrtc
+
+import (
+	"context"
+	"crypto"
+	"encoding/hex"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/p2p/transport/webrtc/udpmux"
+
+	tpt "github.com/libp2p/go-libp2p/core/transport"
+	ma "github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr/net"
+	"github.com/multiformats/go-multibase"
+	"github.com/multiformats/go-multihash"
+
+	"github.com/pion/ice/v2"
+	"github.com/pion/webrtc/v3"
+)
+
+var _ tpt.Listener = &listener{}
+
+const (
+	candidateSetupTimeout                = 20 * time.Second
+	DefaultMaxInFlightConnections uint32 = 10
```

Comment:
why you assign a type to your constant by the way, AFAIK that is not needed in Go, as it will adapt to wherever you use it?